test-app/tests/integration/components/checkbox-test.gts

```diff
@@ -95,6 +107,43 @@ module('Integration | Component | Checkbox', function (hooks) {
     assert.dom('[data-checkbox]').hasNoClass('bg-primary-idle');
   });
 
+  test('it applies the expected classes when `@isChecked={{true}}` and `@isReadOnly={{true}}', async function (assert) {
+    await render(<template>
+      {{! we do not require a label, but instead suggest using Field / TextareaField }}
+      {{! template-lint-disable require-input-label }}
+      <CheckboxControl @isChecked={{true}} @isReadOnly={{true}} data-checkbox />
+    </template>);
+
+    assert.dom('[data-checkbox]').hasClass('bg-titles-and-attributes');
```

Comment:
Very thorough! It's too bad for cases like this you can't pass an array of classes in to check for...